inference/server/oasst_inference_server/routes/chats.py

```diff
@@ -21,12 +22,47 @@
 async def list_chats(
     include_hidden: bool = False,
     ucr: UserChatRepository = Depends(deps.create_user_chat_repository),
+    limit: int | None = Query(10, gt=0, le=100),
+    after: str | None = None,
+    before: str | None = None,
 ) -> chat_schema.ListChatsResponse:
     """Lists all chats."""
     logger.info("Listing all chats.")
-    chats = await ucr.get_chats(include_hidden=include_hidden)
+
+    def encode_cursor(chat: models.DbChat):
+        return base64.b64encode(chat.id.encode()).decode()
+
+    def decode_cursor(cursor: str | None):
+        if cursor is None:
+            return None
+        return base64.b64decode(cursor.encode()).decode()
```

Comment:
why do we base64 encode / decode the ids?

Reply:
Just to hide sensitive information from the eyes. Also it will prevent us from problems if the frontend forgot to url encode the cursor, not really necessary in this case.